src/source/source_simple_transistor.ts

```diff
@@ -0,0 +1,40 @@
+import { z } from "zod"
+import {
+  source_component_base,
+  type SourceComponentBase,
+} from "src/source/base/source_component_base"
+import { voltage, current } from "src/units"
+import { expectTypesMatch } from "src/utils/expect-types-match"
+
+export const source_simple_transistor = source_component_base.extend({
+  ftype: z.literal("simple_transistor"),
+  transistorType: z.enum(["NPN", "PNP"]),
+  pinLabels: z.object({
+    emitter: z.string(),
+    base: z.string(),
+    collector: z.string(),
+  }),
```

Comment:
I don't think we will need pinLabels separately. We can get the labels from the `schematic_port`

Reply:
done.